### PR TITLE
Use HikariCP as JDBC connection pool

### PR DESCRIPTION
**Pull request by MarinHadzhiev**
Replace current implementation of JDBC connection pool with [HikariCP](https://github.com/brettwooldridge/HikariCP). Pool configurations can be changed with env variables - <db_name>\_HIKARI\_<[hikari_prop_name](https://github.com/brettwooldridge/HikariCP#gear-configuration-knobs-baby)>.

Examples:
HANA_HIKARI_dataSource.cachePrepStmts=true
or
HANA_HIKARI_leakDetectionThreshold="1000"

### CODE REVIEW

File: modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java [hunk 1]
```diff
@@ -58,16 +58,17 @@
...
     private final OData2EventHandler odata2EventHandler;
     private DataSource dataSource;
     private final ResultSetReader resultSetReader;
 
-    private static final String DUMMY_BUILDER = "dummyBuilder";
-    private static final String SELECT_BUILDER = "selectBuilder";
-    private static final String INSERT_BUILDER = "insertBuilder";
-    private static final String UPDATE_BUILDER = "updateBuilder";
-    private static final String SQL_CONTEXT = "sqlContext";
+    private static final String DUMMY_BUILDER_CONTEXT_KEY = "dummyBuilder";
```
Comment (delchev) [inline modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java:67]: can we reverse DUMMY_BUILDER_CONTEXT_KEY -> CONTEXT_KEY_DUMMY_BUILDER to become common prefix, not common suffix

**Review by delchev — CHANGES_REQUESTED** [review]

File: modules/database/database-custom/src/main/java/org/eclipse/dirigible/database/custom/CustomDatabase.java [hunk 2]
```diff
@@ -102,27 +105,40 @@ private DataSource initializeDataSource(String name) {
...
-				basicDataSource.setConnectionProperties(databaseConnectionProperties);
-			}
-			WrappedDataSource wrappedDataSource = new WrappedDataSource(basicDataSource);
+		if ((databaseUrl != null) && (databaseUsername != null) && (databasePassword != null)) {
+			Properties props = new Properties();
+			props.setProperty("driverClassName", databaseDriver );
+			props.setProperty("dataSource.user", databaseUsername );
+			props.setProperty("dataSource.password", databasePassword );
+			props.setProperty("jdbcUrl", databaseUrl);
+			props.setProperty("leakDetectionThreshold", "10000" );
```
[the remark this answers is not in the document]
Comment (MarinHadzhiev) [reply modules/database/database-custom/src/main/java/org/eclipse/dirigible/database/custom/CustomDatabase.java:114]: everything is configurable, this is just a default value, if you provide them as env vars they will be overwritten.

[the remark this answers is not in the document]
Comment (delchev) [reply modules/database/database-custom/src/main/java/org/eclipse/dirigible/database/custom/CustomDatabase.java:114]: now I saw it, thanks!

**Review by delchev — APPROVED** [review]

File: modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java (hunk 1 above)
Comment (MarinHadzhiev) [reply modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java:67]: we can, but the idea to be a suffix is that you can easily type it and autocomplete it and in the same time you have the information where this constant is used - as a key in the context. But if you prefer prefix, I am also fine with it.

**Pull request merged**